### PR TITLE
Make security.encoder_factory public again

### LINKED ISSUE #3944: The service "security.encoder_factory" is not public anymore
### Affected version(s)

4.13-RC1

### Description

`$encoder = System::getContainer()->get('security.encoder_factory')->getEncoder(User::class);`

leads to 

```
"The "security.encoder_factory" service or alias has been removed or inlined when the container was compiled. You should either make it public, or stop using the container directly and use dependency injection instead."
```

See https://contao.slack.com/archives/CK4J0KNDB/p1642491533041100

### PR DESCRIPTION
**Pull request by bytehead**
Fixes #3944

### CODE REVIEW

**Review by m-vo — DISMISSED** [review]

**Review by aschempp — DISMISSED** [review]

**Comment by bytehead** [comment]
Failing test seems unrelated.

**Review by fritzmg — DISMISSED** [review]

**Review by leofeyer — DISMISSED** [review]

File: core-bundle/src/DependencyInjection/Compiler/MakeServicesPublicPass.php [hunk 1]
```diff
@@ -33,8 +33,8 @@ class MakeServicesPublicPass implements CompilerPassInterface
         'security.authentication.trust_resolver',
         'security.firewall.map',
         'security.helper',
-        'security.logout_url_generator',
         'security.encoder_factory',
+        'security.logout_url_generator',
```
Comment (bytehead) [inline core-bundle/src/DependencyInjection/Compiler/MakeServicesPublicPass.php:37]: Would the service (`security.encoder_factory`) correctly sorted not sit after `security.authentication.trust_resolver`?

Comment (bytehead) [reply core-bundle/src/DependencyInjection/Compiler/MakeServicesPublicPass.php:37]: Like in f0e377785?

Comment (leofeyer) [reply core-bundle/src/DependencyInjection/Compiler/MakeServicesPublicPass.php:37]: Yes. 🙈

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @bytehead.

**Pull request merged**